### DISCUSSION TITLE
Split stock by user group by using Project Codes also on Stock and/or Part level

### DISCUSSION THREAD

**Discussion by wieselukas** [inventree/InvenTree#7718] [Ideas]
Hello there,

firstly thanks again to everybody working on Inventree, it's awesome to have this tool available and is mostly working very nicely for us!

We recently got the request from another team in our group which wanted to join our Inventree instance to also track their Stock. We can of course split parts by category or stock by stock locations, but ideally we would like to hide one groups parts to the other group, mainly to not have to ask the question "is this even our part?" when finding something via the search. I am not aware that this could be realized in Inventree, so we are now considering separate instances per group. Having everything in one instance however of course eases maintenance and also offers the chance to share certain parts with other groups.

One suggestion to implement such a feature would be to enhance the Project Codes functionality, as this already allows to assign responsible user groups. This would roughly (from a non-inventree-developers perspective) involve the following:
- provide additional settings to the project codes feature like: apply_project_code_to_stock_entries, apply_project_code_to_parts, hide_stock_entries_to_other_projects, hide_parts_to_other_projects, use_project_as_default_for_selected_user, ..
- introduce the project codes field for parts and stock entries
- automatically filter parts and stock entries to the currently logged in users projects in searches. lists and drop-downs
- manufacturers and suppliers imho could always be shared, but when implementing the feature consistently also build orders, purchase orders and sales orders would have to be filtered. (even though this wouldn't matter much for our use case)

So my questions are:
1. is there maybe a functionality that I overlooked, that implements my desired behavior?
2. if not, is there maybe an open issue that tries to achieve similar behavior?
3. would the suggested implementation fit into Inventree's architecture or is this something that would disrupt current behavior too much?

All the best
Lukas

**Comment by matmair** [comment]
Hi there @wieselukas!

Glad to hear you are enjoying the software.

Your request sounds like it could be achieved by https://github.com/inventree/InvenTree/issues/7466 - can you please read through the discussion there and report there or here back?
From a dev/maintenance standpoint, there are quite a lot of changes and testing required if implemented as requested in this discussion. So this would probably have to be split up, described and challenged if not solvable via permissions. Creating a new solutions where widely-adopted ones can be found in comparable products is often something we end up refatoring after a while.

Specific answers:
1) We have stock ownership - but that only works for stock and has its shortcomings in diverse organisations due to being limited to 1 group owning certain stock https://docs.inventree.org/en/stable/stock/owner/#stock-ownership
2) I think https://github.com/inventree/InvenTree/issues/7466 could be applicable if my understanding of the need is correct
3) I certainly see this being a common requirement in academics, R&D orgs and larger SMEs so I think yes

My opinion is one of many, maybe someone else from @inventree/triage @inventree/maintainer will voice theirs in the next few days

**Comment by wieselukas** [comment]
Hi @matmair!

Thanks for pointing me to the stock ownership feature, this is much closer to my needs than what the project codes offer. Still my ideal scenario would be the possibility to completely hide parts that don't belong to my group. But i think this could be something #7466 could be aiming for.

I will write my user story into #7466. I do like the approach to think it through for multiple scenarios and to have one clear concept of ownership/permissions for all levels.

**Reply by zly20129** [reply]
Hi @wieselukas !
Do your two groups of users need to share any parts?
If so, then I don't have any good insights.
But if not, then I suggest you deploy two servers to serve two groups separately.
If you only have one physical server, you can try running two InvenTree servers on one machine using virtual machines, with different IP addresses.
At least this is the quickest solution to your current needs.

**Reply by wieselukas** [reply]
This is what we will probably do for now. As said in my initial comment it would be nice though to share certain parts and to not have to run several instances. But the options to share parts is just a bonus, so we will probably go the pragmatic route of just hosting two instances until #7466 got implemented.

**Reply by matmair** [reply]
thank you, i will close this discussion so that there is 1 place to discuss how permissions can support your use case

**Reply by zly20129** [reply]
> This is what we will probably do for now. As said in my initial comment it would be nice though to share certain parts and to not have to run several instances. But the options to share parts is just a bonus, so we will probably go the pragmatic route of just hosting two instances until #7466 got implemented.

So it is.
